dockstore/dockstore
# Record entry names in lambda event

**New PR description**
This PR adds the `entryName` and ~`groupId`~ `deliveryId` fields to `LambdaEvent`. I implemented this suggestion https://github.com/dockstore/dockstore/pull/5611#discussion_r1272655096, so that each lambda event now corresponds to a singular entry instead of a GitHub webhook invocation. As a result, each GitHub webhook invocation could create more than one lambda event, depending on the number of entries modified as a result of the webhook invocation. The ~`groupId`~ `deliveryId` field is used to group lambda events that were a result of the same GitHub webhook invocation.

Example: a GitHub release could create two new workflows, workflowA and workflowB. This webhook invocation would result in two new lambda events, one for workflowA and one for workflowB. Both lambda events would have the same ~`groupId`~ `deliveryId` to indicate that they were updated in the same GitHub webhook release.

It appears that the GitHub logs dialog box only shows 100 lambda events. Since this PR increases the number of lambda events created per webhook invocation, I created a new ticket to properly paginate the lambda events in the GitHub logs similar to https://dockstore.org/workflows so the user can search and view all events https://github.com/dockstore/dockstore/issues/5617

Note to self:
- [x] Create ticket to paginate GitHub app logs and allow for search once the PR is merged
- [x] Create ticket to store `X-GitHub-Delivery` in `deliveryId` column

**Screenshots**
Note: this is **not** the final UI change and was quickly thrown together to display the group ID and entry name for reviewers to look at. This ticket will implement the UI changes https://ucsc-cgl.atlassian.net/browse/DOCK-2407
**EDIT:** I ended up renaming the groupId column to deliveryId and that isn't reflected in the screenshots below, but the concept is the same.

_Lambda events for a GitHub installation. Note how the GitHub releases performed during the installation have the same group ID as the installation event because they were released as a result of the installation occuring._
![image](https://github.com/dockstore/dockstore/assets/25287123/d5d5fe63-4044-4c1c-a1a8-61c62c23e866)

_Lambda events for a GitHub release, showing the error message for a workflow that failed._
![image](https://github.com/dockstore/dockstore/assets/25287123/52bb4f66-9930-48c1-b503-64a116ccede0)

_Lambda events for a GitHub release, showing the message for a workflow that was successful but failed validation._
![image](https://github.com/dockstore/dockstore/assets/25287123/67cc9a35-2fa0-4e65-8d47-589ae38a6829)

<details>
<summary>Old PR description</summary>

**Description**
This PR records the entry names associated with the lambda event. Successful events will record all entry names, while failed events will only record entry names that are associated with the failure:
- Installation events: no entry names associated with this type of event
- Push events: records all entry names for a successful event or entries that failed to be created for a failed event
- Publish events: records the entry name associated with the publish event
- Delete events: records the entry names of the entries with versions that were deleted

I modified the GitHub app release validation and error messages so references to the workflow name are consistent. Below are screenshots showing the difference between the messages before and after. I highlighted the workflow names references to show that the after only has the workflow names to match with the soon-to-be entry names column.

**Before**:
![image](https://github.com/dockstore/dockstore/assets/25287123/e1eabc6e-c5ad-4137-8d82-f75016d32fab)

**After**:
Note that I just added the entry names column to the table for easy viewing when testing locally so it's not production-ready. This ticket and PR does not modify the UI (there's a separate ticket for that).
![image](https://github.com/dockstore/dockstore/assets/25287123/cb42f12d-2bfd-474a-aca3-c9116275367c)

</details>

**Misc change:**
While writing the tests, I noticed that the endpoint that installs the GitHub app, `handleGitHubInstallation`, throws an exception if the GitHub releases of the few branches it finds is unsuccessful. According to this [line](https://github.com/dockstore/dockstore/blob/ac5818ca0a1e3b0326f96cb25afab9ae104bdf18/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L2150C26-L2150C26), we specify that it shouldn't throw exceptions if the GitHub release was unsuccessful. It was a simple fix (and I needed it to write my test), so I added it to this PR.


**Review Instructions**
May be easier to review once the UI portion is complete in https://ucsc-cgl.atlassian.net/browse/DOCK-2407
- Perform various GitHub actions (push, delete, installation, publish) for a GitHub apps workflow and verify that the GitHub app logs record the entry name for the lambda events. Verify that a push of a .dockstore.yml with more than 1 workflow results in more than one lambda event, one for each workflow.
- Below is how the entry names are recorded:
  - Installation events: no entry names associated with this type of event
  - Push events: records the entry name for the workflow affected by the push event
  - Publish events: records the entry name associated with the publish event
  - Delete events: records the entry name of the entries with versions that were deleted 

**Issue**
[SEAB-5659](https://ucsc-cgl.atlassian.net/browse/SEAB-5659)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!



[SEAB-5659]: https://ucsc-cgl.atlassian.net/browse/SEAB-5659?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
